### PR TITLE
Change error to warning when creating symlink, that already exists

### PR DESCRIPTION
**Pull request by IncubusRK**
### Database name
PostgreSQL

# Pull request description
Fix error when fetching incremental backup of posgresql database, that have multiple table spaces.

### Describe what this PR fix
wal-g trying to create symlink for additional tablespace on every part of backup and exit witch error

Also see
https://github.com/wal-g/wal-g/issues/585
https://github.com/wal-g/wal-g/issues/499

### Please provide steps to reproduce (if it's a bug)
create new postgres database
create additional table space
push-backup
fetch-backup

### CODE REVIEW

File: internal/databases/postgres/backup.go [hunk 1]
```diff
@@ -235,7 +235,14 @@ func setTablespacePaths(spec TablespaceSpec) error {
 	for _, location := range spec.tablespaceLocations() {
 		err := fs.NewFolder(location.Location, "").EnsureExists()
 		if err != nil {
-			return fmt.Errorf("error creating folder for tablespace %v", err)
+			// Don't show error when correct symlink already exist
```
Comment (x4m) [inline internal/databases/postgres/backup.go:238]: it would be very good to do this check within EnsureExists() function.

**Review by x4m — COMMENTED** [review]
Many thanks for fixing this bug! Your fix seems correct. But it would be more robust to refactor this stuff a little. What if we do this check in the EnsureExists() function? just check there that err is "Dir already exists"?

Thank you!

File: internal/databases/postgres/backup.go (hunk 1 above)
Comment (IncubusRK) [reply internal/databases/postgres/backup.go:240]: EnsureExists() used for Folder type. 
I think, I should not create a new type to work with symbolic links if only a few calls are needed.

**Comment by IncubusRK** [comment]
This fix only help to prevent canceling of backup-fetch process. 
May be also need to change the recovery logic.

For example after run next command in /data (original base also backuped from /data) dir:
wal-g backup-fetch ./backup-test LATEST
I got /data/backup-test with recovered database and /data/pg_tblspc with empty directories and sysmlinks to it.
It meants that recovery process use relative path to recovery additional tablespaces, which may not be convenient in terms of free space for a very large database.

**Comment by usernamedt** [comment]
> This fix only help to prevent canceling of backup-fetch process. May be also need to change the recovery logic.
> 
> For example after run next command in /data (original base also backuped from /data) dir: wal-g backup-fetch ./backup-test LATEST I got /data/backup-test with recovered database and /data/pg_tblspc with empty directories and sysmlinks to it. It meants that recovery process use relative path to recovery additional tablespaces, which may not be convenient in terms of free space for a very large database.

I guess it would be cool to add the check for the correct handling of delta backups to this [integration test](https://github.com/wal-g/wal-g/blob/master/docker/pg_tests/scripts/tests/tablespace_support_test.sh) to make sure it won't accidentally break in the future.

File: internal/databases/postgres/backup.go (hunk 1 above)
Comment (usernamedt) [reply internal/databases/postgres/backup.go:240]: You can add this logic to the [TablespaceLocation](https://github.com/wal-g/wal-g/blob/7f699a029a4d625c18136ebf9e9d87b022c27022/internal/databases/postgres/tablespace_spec.go#L28) type

**Comment by IncubusRK** [comment]
Added some minor fixes for passing linter checks

**Review by serprex — APPROVED** [review]

**Pull request merged**